src/core/RecyclerListView.tsx

```diff
@@ -300,7 +300,11 @@ export default class RecyclerListView<P extends RecyclerListViewProps, S extends
 
     public getCurrentScrollOffset(): number {
         const viewabilityTracker = this._virtualRenderer.getViewabilityTracker();
-        return viewabilityTracker ? viewabilityTracker.getLastActualOffset() + this.props.distanceFromWindow! : 0;
+        let currentOffset = (viewabilityTracker ? viewabilityTracker.getLastActualOffset() : 0);
+        if (currentOffset > this.props.distanceFromWindow) {
```

Comment:
While this will fix the specific issue internally, this doesn't look correct. An example would be having scrolled very little such that it is < distanceFromWindow. We will get 0 as the current offset which is incorrect because there indeed is a valid scroll offset